source/common/network/dns_impl.cc

```diff
@@ -169,20 +169,18 @@ void DnsResolverImpl::PendingResolution::onAresGetAddrInfoCallback(int status, i
 
   if (completed_) {
     if (!cancelled_) {
-      // TODO(chaoqin-li1123): remove this exception catching by refactoring.
-      //  We can't add a main thread assertion here because both this code is reused by dns filter
-      //  and executed in both main thread and worker thread. Maybe split the code for filter and
-      //  main thread.
-      TRY_NEEDS_AUDIT { callback_(resolution_status, std::move(address_list)); }
-      catch (const EnvoyException& e) {
+      // Use a raw try here because it is used in both main thread and filter.
+      // Can not convert to use status code as there may be unexpected exceptions in server fuzz
+      // tests, which must be handled.
```

Comment:
Can we leave this as a TODO?

I think we just need to understand other ways the server-test might throw unexpectedly and fix them.

Reply:
I saw that we are parsing some ports inside the callback, maybe that exception is no fixable.

Reply:
Can you gather all the info and leave it here as a TODO?

Reply:
Thanks, will do.